libpdbg/kernel.c

```diff
@@ -79,7 +79,7 @@ static int kernel_getscom(struct target *target, uint64_t addr, uint64_t *value)
 static int kernel_fsi_getcfam(struct target *target, uint64_t addr64, uint64_t *value)
 {
 	int rc;
-	uint32_t addr = (addr64 & 0xffff00) | ((addr64 & 0xff) << 2);
+	uint32_t addr = (addr64 & 0xfc00) | ((addr64 & 0x3ff) << 2);
 
```

Comment:
This is masking off too many upper bits. For secondary processor access to work we need to be able to pass addresses up to 0x480000. So the patch should be (addr64 & 0x7ffc00) | ((addr64 & 0x3ff) << 2).

Also we have the same bug in bmcfsi.c - can you please add that to the patch (although in this case you *do* need to mask off some of the higher order bits as they go in the slave-id for that backend). Thanks!